src/main/java/bot/Sailfish.java

```diff
@@ -0,0 +1,198 @@
+package bot;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The bot.
+ */
+public final class Sailfish {
+    /**
+     * The width for the screen.
+     */
+    private static final int WIDTH = 20;
+
+    /**
+     * Scanner object provided to the bot.
+     */
+    private final Scanner scanner;
+
+    /**
+     * The list of tasks stored by the bot.
+     */
+    private final List<Task> tasks;
+
+    /**
+     * Creates the bot.
+     */
+    public Sailfish() {
+        this.scanner = new Scanner(System.in);
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Makes the bot take control of the application.
+     * <p>
+     * This method shows the welcome message and then waits for input.
+     */
+    public void takeControl() {
+        // Prints the welcome text.
+        this.printWelcome();
+
+        // Input loop.
+        while (true) {
+            System.out.println("-".repeat(WIDTH));
+            // Get input from the user.
+            Command command = Command.fromString(this.scanner.nextLine());
+            System.out.println("-".repeat(WIDTH));
+
+            // Switch the command.
+            switch (command.getCommand()) {
+            case "list": // List all tasks.
+                this.list();
+                break;
+            case "mark": // Mark a task as done.
+                this.mark(command);
+                break;
+            case "unmark": // Mark a task as not done.
+                this.unMark(command);
+                break;
+            case "bye": // Exit the app.
+                System.out.println("Farewell, sailor!");
+                return;
+            default: // By default, we assume that the user is using a command that adds a task.
+                this.addTask(command);
```

Comment:
Good idea to throw adding together here and determine difference using separate functions